### PR TITLE
refactor(apig): Refactor application resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Refactor application resource:
- rewrite methods.
- deprecate 'update_time' parameter.
- rename the 'registraion_time' parameter.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor application resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApplication_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApplication_basic -timeout 360m -parallel 4
=== RUN   TestAccApplication_basic
=== PAUSE TestAccApplication_basic
=== CONT  TestAccApplication_basic
--- PASS: TestAccApplication_basic (494.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      494.294s
```

### CODE REVIEW

File: huaweicloud/services/apig/resource_huaweicloud_apig_application.go [hunk 1]
```diff
@@ -214,106 +230,112 @@ func isCodeInApplication(codes []applications.AppCode, code string) (string, boo
...
-func removeApigV2ApplicationCode(d *schema.ResourceData, client *golangsdk.ServiceClient,
-	results []applications.AppCode, code string) error {
-	instanceId := d.Get("instance_id").(string)
-	id, ok := isCodeInApplication(results, code)
-	if !ok {
-		return fmtp.Errorf("Code is not in the application (%s)", d.Id())
+func removeApplicationCodes(client *golangsdk.ServiceClient, instanceId, appId string, codes []interface{}) error {
+	results, err := queryApplicationCodes(client, instanceId, appId)
+	if err != nil {
+		return fmtp.Errorf("error retrieving application codes: %s", err)
```
Comment (houpeng80) [inline huaweicloud/services/apig/resource_huaweicloud_apig_application.go]: replaced fmtp with diag

File: huaweicloud/services/apig/resource_huaweicloud_apig_application.go [hunk 2]
```diff
@@ -1,208 +1,224 @@
...
 	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/apigw/dedicated/v2/applications"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
```
Comment (houpeng80) [inline huaweicloud/services/apig/resource_huaweicloud_apig_application.go]: remove this import

Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_application.go]: fixed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_application.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_application.go]: actually is fmt

File: docs/resources/apig_application.md [hunk 3]
```diff
@@ -26,45 +26,54 @@ resource "huaweicloud_apig_application" "test" {
...
 
 ## Import
 
-APIG Applications can be imported using their `id` and ID of the APIG dedicated instance to which the application
-belongs, separated by a slash, e.g.
+Applications can be imported using their `id` and the ID of the related dedicated instance, separated by a slash, e.g.
 
 ```
-$ terraform import huaweicloud_apig_application.test <instance id>/<id>
+$ terraform import huaweicloud_apig_application.test &ltinstance_id&gt/&ltid&gt
```
Comment (ShiChangkuo) [inline docs/resources/apig_application.md]:
```suggestion
```shell
$ terraform import huaweicloud_apig_application.test <instance_id>/<id>
```
```

Comment (Lance52259) [reply docs/resources/apig_application.md]: done.

**Pull request merged**